### PR TITLE
Clarify and reorganize modular repository

### PR DESCRIPTION
**Pull request by breakdancingcat**
- Breaking apart how components are configured to better understand and troubleshoot

### CODE REVIEW

File: app/Repositories/ModularPageRepository.php [hunk 1]
```diff
@@ -287,4 +276,78 @@ public function organizePromoItemsByOption(array $data)
...
+        // Set hero from components
+        $hero = collect($promos['components'])->reject(function ($data, $component_name) {
+            return !str_contains($component_name, 'hero');
+        })->toArray();
+
+        if (!empty($hero)) {
+            $hero_key = array_key_first($hero);
+
+            // Minimal hero overwrites any other hero data
+            if ($hero_key === 'misb-minimal-hero' && !empty($promos['components'][$hero_key]['component']['filename'])) {
```
Comment (nickdenardis) [inline app/Repositories/ModularPageRepository.php:294]: Perhaps this isn't needed in this project? 
'misb-minimal-hero'

File: app/Repositories/ModularPageRepository.php [hunk 2]
```diff
@@ -230,7 +170,6 @@ public function getPromos($components, $site_id)
             // Adjust promo data
             $data = collect($data)->map(function ($item) use ($components, $name) {
                 $item = $this->adjustPromoData($item, $components['components'][$name]);
```
Comment (nickdenardis) [inline app/Repositories/ModularPageRepository.php:172]: Return directly here instead of making a temp variable?

**Review by chrispelzer — APPROVED** [review]

File: app/Repositories/ModularPageRepository.php (hunk 1 above)
Comment (breakdancingcat) [reply app/Repositories/ModularPageRepository.php:294]: Ty! I'm sorry I missed that

**Review by nickdenardis — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**